src/interceptors/Socket/MockSocket.test.ts

```diff
@@ -150,43 +165,74 @@ it('calls the "read" on "socket.read(null)"', () => {
   expect(readCallback).toHaveBeenCalledWith(null, undefined)
 })
 
-it('updates the readable/writable state on "socket.end()"', async () => {
+it('updates the writable state on "socket.end()"', async () => {
+  const finishListener = vi.fn()
+  const endListener = vi.fn()
   const socket = new MockSocket({
     write: vi.fn(),
     read: vi.fn(),
   })
+  socket.on('finish', finishListener)
+  socket.on('end', endListener)
 
   expect(socket.writable).toBe(true)
   expect(socket.writableEnded).toBe(false)
   expect(socket.writableFinished).toBe(false)
-  expect(socket.readable).toBe(true)
-  expect(socket.readableEnded).toBe(false)
 
   socket.write('hello')
+  // Finish the writable stream.
   socket.end()
 
   expect(socket.writable).toBe(false)
   expect(socket.writableEnded).toBe(true)
-  expect(socket.readable).toBe(true)
 
+  // The "finish" event is emitted when writable is done.
+  // I.e. "socket.end()" is called.
   await vi.waitFor(() => {
-    socket.once('finish', () => {
-      expect(socket.writableFinished).toBe(true)
-    })
+    expect(finishListener).toHaveBeenCalledTimes(1)
+  })
+  expect(socket.writableFinished).toBe(true)
+})
+
+it('updates the readable state on "socket.push(null)"', async () => {
+  const endListener = vi.fn()
+  const socket = new MockSocket({
+    write: vi.fn(),
+    read: vi.fn(),
   })
+  socket.on('end', endListener)
+
+  expect(socket.readable).toBe(true)
+  expect(socket.readableEnded).toBe(false)
+
+  socket.push('hello')
+  socket.push(null)
+
+  expect(socket.readable).toBe(true)
+  expect(socket.readableEnded).toBe(false)
+
+  // Read the data to free the buffer and
+  // make Socket emit "end".
+  socket.read()
```

Comment:
This was a broken test. We have to call `.read()` to exhaust the readable stream so it emits the `end` event.